examples/skybox/main.rs

```diff
@@ -265,11 +267,14 @@ impl framework::Example for Skybox {
         // update rotation
         let rotation = cgmath::Matrix4::<f32>::from_angle_x(cgmath::Deg(0.25));
         self.uniforms[1] = self.uniforms[1] * rotation;
-        queue.write_buffer(
+        let raw_uniforms = raw_uniforms(&self.uniforms);
+        self.staging_belt.write_buffer(
             &self.uniform_buf,
             0,
-            bytemuck::cast_slice(&raw_uniforms(&self.uniforms)),
-        );
+            wgpu::BufferSize::new((raw_uniforms.len() * 4) as wgpu::BufferAddress).unwrap(),
+            device,
+        ).copy_from_slice(bytemuck::cast_slice(&raw_uniforms));
+        self.staging_belt.flush(queue, device);
```

Comment:
If the future returned by `StagingBelt::flush` isn't polled, the buffers won't be mapped again, right?

Reply:
Right. This is incomplete, I got stuck on figuring out the futures interaction with the event loop.

Reply:
actually, your concern still stands. Even with the current code, the poll() isn't happening.